internal/pkg/deploy/cloudformation/project.go

```diff
@@ -221,6 +222,79 @@ func (cf CloudFormation) AddEnvToProject(project *archer.Project, env *archer.En
 	return nil
 }
 
+var getRegionFromClient = func(client cloudformationiface.CloudFormationAPI) (string, error) {
+	concrete, ok := client.(*cloudformation.CloudFormation)
+	if !ok {
+		return "", errors.New("failed to retrieve the region")
+	}
+	return *concrete.Client.Config.Region, nil
+}
+
+// AddPipelineResourcesToProject adds resources needed to support a pipeline
+// to a project. It conditionally creates resources needed for the pipeline
+// in a AWS region in case there's no existing environment
+// (which would have provisioned these resources) for the project
+func (cf CloudFormation) AddPipelineResourcesToProject(project *archer.Project, pipelineName string) error {
```

Comment:
We can get rid of the pipelineName and I think maybe I prefer setting the region as a parameter, rather than pulling it from the client.